services/harmony/app/frontends/ogc-edr/get-data-for-area.ts

```diff
@@ -0,0 +1,131 @@
+import { createDecrypter, createEncrypter } from '../../util/crypto';
+import { handleCrs, handleExtend, handleFormat, handleGranuleIds, handleGranuleNames, handleScaleExtent, handleScaleSize } from '../../util/parameter-parsers';
+import { keysToLowerCase } from '../../util/object';
+import { ParameterParseError, mergeParameters, parseWkt } from '../../util/parameter-parsing-helpers';
+import { parseSubsetParams } from '../ogc-coverages/util/subset-parameter-parsing';
+import { Response, NextFunction } from 'express';
+import DataOperation from '../../models/data-operation';
+import env from '../../util/env';
+import HarmonyRequest from '../../models/harmony-request';
+import wrap from '../../util/array';
+import { RequestValidationError } from '../../util/errors';
+import { parseVariables } from '../../util/variables';
+import { parseDatetime } from './util/helper';
+
+/**
+ * Express middleware that responds to OGC API - EDR Area GET requests.
+ * Responds with the actual EDR data.
+ *
+ * This function merely sets up a query and proxies the request to the `getDataForArea`
+ * function.
+ *
+ * @param req - The request sent by the client
+ * @param res - The response to send to the client
+ * @param next - The next express handler
+ */
+export function getDataForArea(
```

Comment:
nit: Most of the code except `coords` in this function is duplicate of `getDataForCube`. We should refactor this common code out and share between these two functions.

Reply:
Done.